### PR TITLE
Show keyboard shortcuts in Workbench editor's Options menu

### LINKED ISSUE #24993: Keyboard shortcuts for workbench editor
**Original reporter:** [username facility]/[nobody]. 

### Expected behavior
The keyboard shortcuts for Run should be visible somewhere in the workbench.

Also it would be good to add Run All ctrl + shift + enter as it worked in mantidplot.

### Actual behavior
ctrl+enter works for run, but is not listed in the menu item or tooltip.

### PR DESCRIPTION
**Pull request by dtasev**
**Description of work.**
It seems Qt 5.10 decided that all QMenus not inside a QMenuBar are context menus, relevant Qt change: https://codereview.qt-project.org/#/c/219931/.

This PR makes the keyboard shortcuts visible inside the Options menu on the Workbench editor again.

**To test:**
Open Workbench, click the `Options` menu, shortcuts should be displayed inside

Fixes #24993

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/utils/qt/__init__.py [hunk 1]
```diff
@@ -147,6 +151,10 @@ def create_action(parent, text, on_triggered=None, shortcut=None,
     if icon_name is not None:
         action.setIcon(get_icon(icon_name))
 
+    # shortcuts in context menus option is only available after Qt 5.10
+    if 5 == QT_VERSION_INT[0] and 10 <= QT_VERSION_INT[1] and shortcut_visible_in_context_menu:
```
Comment (martyngigg) [inline qt/python/mantidqt/utils/qt/__init__.py]: What about the inevitable Qt 6? Would a `hasattr` check on `action` be better?

File: qt/widgets/common/inc/MantidQtWidgets/Common/ManageUserDirectories.h [hunk 2]
```diff
@@ -18,7 +18,8 @@ class EXPORT_OPT_MANTIDQT_COMMON ManageUserDirectories : public QDialog {
   Q_OBJECT
 
 public:
-  ManageUserDirectories(QWidget *parent = nullptr);
```
Comment (martyngigg) [inline qt/widgets/common/inc/MantidQtWidgets/Common/ManageUserDirectories.h]: Have these changes been put in this PR by accident? They don't seem relevant to the description.

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: qt/python/mantidqt/utils/qt/__init__.py (hunk 1 above)
Comment (DTasev) [reply qt/python/mantidqt/utils/qt/__init__.py]: Good call!

File: qt/widgets/common/inc/MantidQtWidgets/Common/ManageUserDirectories.h (hunk 2 above)
Comment (DTasev) [reply qt/widgets/common/inc/MantidQtWidgets/Common/ManageUserDirectories.h]: 👍 	This is work from #24783, must have added the files at some point by accident

**Review by martyngigg — APPROVED** [review]
Looks good. New run all option works as expected.
